src/shared/functions.py

```diff
@@ -264,3 +265,56 @@ def get_latest_data_package_version(bucket, prefix):
     if "Contents" not in s3_res or highest_ver is None:
         logger.error("No data package versions found for %s", prefix)
     return highest_ver
+
+
+@dataclasses.dataclass()
+class PackageMetadata:
+    study: str
+    site: str | None
+    data_package: str
+    version: str
+
+
+def parse_s3_key(key: str) -> dict:
+    """Handles extraction of package metadata from an s3 key"""
+    key = key.split("/")
+    print(key)
+    match key[0]:
+        case enums.BucketPath.AGGREGATE.value | enums.BucketPath.CSVAGGREGATE.value:
+            package = PackageMetadata(
+                study=key[1],
+                site=None,
+                data_package=key[2].split("__")[1],
+                version=key[3],
+            )
+        case (
+            enums.BucketPath.ARCHIVE.value
+            | enums.BucketPath.CSVFLAT.value
+            | enums.BucketPath.FLAT.value
+            | enums.BucketPath.LATEST.value
+        ):
+            package = PackageMetadata(
+                study=key[1],
+                site=key[3],
+                data_package=key[2].split("__")[1],
+                version=key[4],
+            )
+        case enums.BucketPath.ERROR.value | enums.BucketPath.LAST_VALID.value:
```

Comment:
This branch looks identical to the above branch. Any reason to duplicate?

Reply:
so it is! it didn't start out that way, but will consolidate.